### PR TITLE
Strict JWT exp/nbf claim validation (Copilot #92 follow-up)

### PR DESCRIPTION
**Pull request by kamrankhan78694**
## Subsystem
`middleware_auth` — JWT claim-validation hardening. Follow-up to the Copilot review on #92 (audit #32).

## Problem
Copilot's review of #92 raised three claim-validation gaps. None is exploitable without the HMAC secret (forging any claim requires re-signing), but each is the wrong behavior for a security-critical field:

1. **Present-but-non-integer claim silently ignored.** `jwt_claim_int` returned a plain `false` both when a key was absent *and* when it was present with a malformed value — so `"exp":"9999999999"` (a quoted string) or `"exp":9999999999abc` (trailing garbage) caused exp/nbf enforcement to be **skipped**, not to fail.
2. **Non-positive exp skipped.** The check `exp_val > 0 && exp_val < now` ignored `exp <= 0`, i.e. a timestamp at/before the epoch (definitely expired) was accepted.
3. **`exp == now` treated as valid.** RFC 7519 §4.1.4 requires the current time to be *before* exp.

## Fix
`jwt_claim_int()` now returns a tri-state:
```c
typedef enum { JWT_CLAIM_ABSENT, JWT_CLAIM_INVALID, JWT_CLAIM_OK } jwt_claim_status_t;
```
A key present with a non-integer value, or an integer followed by anything other than optional whitespace and then a `,` or `}` (a JSON object value terminator), is **INVALID**. `parse_jwt_token()`:
- rejects the token when exp **or** nbf is `INVALID`;
- rejects exp when `now >= exp` (no sign guard — a non-positive exp is at/before the epoch);
- rejects nbf when `now < nbf`;
- keys `require_exp` off `exp_st != OK`.

## Tests
Four validly-HMAC-signed vectors added to `test_jwt_auth_verify` (so only the claim checks can gate them): `JWT_EXP_ZERO` (`exp=0`), `JWT_EXP_STRING` (`exp="9999999999"`), `JWT_EXP_GARBAGE` (`exp=9999999999abc`), `JWT_NBF_STRING` (`nbf="9999999999"`, exp itself valid). All are now rejected; each was accepted before.

**Negative controls:** (a) restoring the `exp_val > 0` skip re-accepts `JWT_EXP_ZERO`; (b) disabling the INVALID-claim rejection re-accepts the malformed-claim vectors. Both confirmed, then restored. All prior vectors (valid / expired / no-exp / future-nbf / HS384 / alg-none / bad-sig / no-colon) still pass.

_Not unit-tested:_ the `exp == now` boundary (item 3) — it needs a token minted at test time with `exp = time(NULL)`, and the library intentionally exposes no JWT-minting helper. `JWT_EXP_ZERO` covers the removal of the sign guard against a fixed boundary.

## Validation
- Normal build: `ctest` **6/6**, zero warnings.
- ASan/UBSan build: `ctest` **6/6**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by kamrankhan78694 — COMMENTED** [review]
## Independent review

Reviewed for RFC 7519 conformance of the exp/nbf logic and correctness of the tri-state claim parse. **No blocking issues.**

### Verified
- **ABSENT vs INVALID is the crux.** Returning `false` for both "absent" and "present-but-malformed" was the root defect; the tri-state now lets `parse_jwt_token` reject a present-but-malformed exp/nbf instead of skipping enforcement. All four malformed vectors (`exp=0`, `exp="9999..."`, `exp=9999...abc`, `nbf="9999..."`) plus `exp=+9999...` are rejected, each with a negative control.
- **Terminator check is JSON-correct.** After the integer, optional whitespace then `,` or `}` is required — handles the last-claim (`}`), mid-object (`,`), and pretty-printed (`123\n}`) cases, and rejects trailing garbage. A leading `+` / lone `-` / quoted string is rejected up front.
- **exp/nbf comparisons match the RFC.** `now >= exp` rejects (RFC §4.1.4 requires now < exp; a non-positive exp is at/before the epoch = expired); `now < nbf` rejects (§4.1.5). No sign guards masking edge values.
- **No behavior change for well-formed tokens.** All prior vectors (valid / expired / no-exp / future-nbf / HS384 / alg-none / bad-sig / no-colon) still pass; `require_exp` keyed off `exp_st != OK` preserves the require-exp semantics.

### Notes (no change requested)
- `strtol` `ERANGE` isn't inspected: an astronomically large exp saturates to `LONG_MAX` (far future → valid) and a large nbf to `LONG_MAX` (→ not-yet-valid → rejected) — both safe.
- The `exp == now` boundary isn't unit-tested (needs a token minted at test time; the library exposes no JWT-minting helper). `JWT_EXP_ZERO` covers the removal of the positive-only guard against a fixed boundary.
- This is a flat-scan claim reader, not a full JSON parser (matches the first real `"exp"`/`"nbf"` key at any nesting) — unchanged from before and adequate for the top-level timing claims.

_🤖 Reviewed by [Claude Code](https://claude.com/claude-code)_

File: src/middleware_auth.c [hunk 1]
```diff
@@ -776,19 +787,28 @@ static bool jwt_claim_int(const char *payload_json, const char *key, long *out)
...
     if (!json_skip_to_value(&v)) {
-        return false;                 /* malformed: no key:value separator */
+        return JWT_CLAIM_INVALID;      /* no key:value separator */
     }
     char *endptr = NULL;
     long val = strtol(v, &endptr, 10);
     if (endptr == v) {
-        return false;                 /* non-numeric value */
+        return JWT_CLAIM_INVALID;      /* not a bare number (e.g. a quoted string) */
+    }
```
[the remark this answers is not in the document]
Comment (kamrankhan78694) [reply src/middleware_auth.c:808]: Fixed in 65206b0 — and this one bites a real target, not just Windows: `middleware_auth.c` is in the core sources (`CMakeLists.txt:46`), so it compiles for **wasm32**, where `long` is 32-bit. `jwt_claim_int` now parses with `strtoll` into `long long` and the exp/nbf comparisons cast `now` to `long long`, so a NumericDate like `9999999999` (year 2286) is enforced identically on LP64, wasm32, and LLP64.

Rather than round-trip-into-`long`, I widened the whole path to `long long` (still C99, no new include) — simpler and avoids rejecting legitimately large-but-valid dates. No behavior change on this LP64 dev/CI box (long is already 64-bit), so ctest stays 6/6; the divergence this fixes only manifests on 32-bit-`long` targets, so it isn't reproducible as a failing unit test here.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/middleware_auth.c [hunk 2]
```diff
@@ -887,21 +913,28 @@ static bool parse_jwt_token(const char *token,
...
+         * quoted string) is rejected outright — never silently treated as absent. */
+        if (exp_st == JWT_CLAIM_INVALID || nbf_st == JWT_CLAIM_INVALID) {
+            goto cleanup;
+        }
+        /* exp: RFC 7519 §4.1.4 requires the current time to be BEFORE exp, so reject
+         * at or after it. No sign guard: a non-positive exp is at/before the epoch,
+         * i.e. already expired. */
+        if (exp_st == JWT_CLAIM_OK && now >= (time_t)exp_val) {
             goto cleanup; /* Token expired */
         }
```
[the remark this answers is not in the document]
Comment (kamrankhan78694) [reply src/middleware_auth.c]: Already addressed by 65206b0 (the strtoll change): the exp/nbf comparisons no longer cast `exp_val`/`nbf_val` to `time_t`. `exp_val`/`nbf_val` are now `long long`, and the checks are `(long long)now >= exp_val` and `(long long)now < nbf_val` — i.e. the comparison happens in signed 64-bit, casting *now* (a small positive timestamp) rather than the attacker-controlled value. A non-positive exp (`-5`, `0`) therefore yields `now >= exp` → true → rejected as expired, with no wrapping. `(long long)now` is safe for any real timestamp (would need `now >= 2^63`, ~292 billion years out).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: tests/test_weblib.c [hunk 3]
```diff
@@ -1968,6 +1968,18 @@ void test_jwt_auth_create_destroy(void) {
  * the "alg" key and its value). A structural parse must reject it; a lenient skip
  * that treats ':' like whitespace would accept it. */
 #define JWT_ALG_NO_COLON "eyJhbGciIkhTMjU2In0.eyJzdWIiOiJ1IiwiZXhwIjo5OTk5OTk5OTk5fQ.e8_VDkko9gFLk-HO1bo1QpuDUilzF66p8A26jdNh6BA"
+/* Validly-signed tokens exercising strict exp/nbf claim validation. Each is rejected
+ * only by the claim checks (alg/signature are fine):
+ *   ZERO    exp=0        -> at/before epoch = expired (no positive-only skip)
+ *   STRING  exp="9999.." -> present but a quoted string, not an integer
+ *   GARBAGE exp=9999..abc-> integer prefix with trailing garbage
+ *   NBFSTR  nbf="9999.." -> present-but-non-integer nbf (exp itself valid/future) */
```
[the remark this answers is not in the document]
Comment (kamrankhan78694) [reply tests/test_weblib.c:1976]: Minor/cosmetic: JWT_EXP_PLUS is documented on its own comment line directly below that four-case block, so the vectors are all described — the list just doesn't repeat it. Noting for a future test touch-up rather than reopening the merged PR.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
